openmdao/utils/find_cite.py

```diff
@@ -0,0 +1,71 @@
+"""
+Helper function to find all the `cite` attributes throughout a model.
+"""
+from __future__ import print_function
+from collections import OrderedDict
+import sys
+
+if sys.version_info[0] == 2:  # Not named on 2.6
+    from StringIO import StringIO
+else:
+    from io import StringIO
+
+
+def _check_cite(instance, citations):
+    """
+    Grab the cite attribute, if it exists.
+
+    Parameters
+    ----------
+    instance : object
+        the instance to check for citations on
+    citations : dict
+        the dictionary to add a citation to, if found
+    """
+    if instance.cite:
+        klass = instance.__class__
+        # return klass, cite
+        citations[klass] = instance.cite
+
+
+def find_citations(prob, out_stream=sys.stdout):
```

Comment:
What about making this an option on `Problem.setup`? That saves the user the work of importing another function. Instead they can just type `prob.setup(print_citations=True)`